tools/update_es_based_on_url_keyword.py

```diff
@@ -0,0 +1,172 @@
+    those that continue to have 'klima' as a keyword. For this reason, 
+    RECALCULATE_TOPICS should be true. This setting MUST BE true, in case of an update,
+    e.g. we adde a new mapping to one of the keywords.
+    - RE_INDEX_ONLY_ARTICLES_IN_ES (bool): if the articles re-index are only those 
+    that were already in ES.
+    - ITERATION_STEP (int): number of articles to index in each loop. 
+"""
+
+URL_KEYWORD_TO_UPDATE = "vejret"
+DELETE_ARTICLE_NEW_TOPICS = True
+RECALCULATE_TOPICS = True
+RE_INDEX_ONLY_ARTICLES_IN_ES = True
+ITERATION_STEP = 1000
+
+
+# coding=utf-8
+from zeeguu.core.elastic.indexing import (
+    create_or_update_bulk_docs,
+)
+
+from elasticsearch import Elasticsearch
+from elasticsearch.helpers import bulk, scan
+import zeeguu.core
+from zeeguu.core.model import Article
+from datetime import datetime
+from sqlalchemy.orm.exc import NoResultFound
+from zeeguu.api.app import create_app
+
+from zeeguu.core.model import (
+    ArticleUrlKeywordMap,
+    UrlKeyword,
+    NewArticleTopicMap,
+    NewTopic,
+)
+from zeeguu.core.model.article import article_topic_map
+from zeeguu.core.elastic.settings import ES_ZINDEX, ES_CONN_STRING
+import numpy as np
+from tqdm import tqdm
+
+app = create_app()
+app.app_context().push()
+
+
+es = Elasticsearch(ES_CONN_STRING)
+db_session = zeeguu.core.model.db.session
+print(es.info())
+
+
+def main():
+    def fetch_articles_by_id(id_list, recalculate_topic=False):
+        for i in id_list:
+            try:
+                article = Article.find_by_id(i)
+                if not article:
+                    print(f"Skipped for: '{i}', article not in DB.")
+                    continue
+                if recalculate_topic:
+                    article.recalculate_topics_from_url_keywords(db_session)
+                    db_session.commit()
+                yield (article)
+            except NoResultFound:
+                print(f"fail for: '{i}'")
+            except Exception as e:
+                print(f"fail for: '{i}', {e}")
+
+    def gen_docs(articles_w_topics):
+        for article in articles_w_topics:
+            try:
+                yield create_or_update_bulk_docs(article, db_session)
+            except Exception as e:
+                print(f"fail for: '{article.id}', {e}")
+
+    # Get the articles for the url_keyword
+    target_ids = np.array(
+        [
+            a_id[0]
+            for a_id in db_session.query(Article.id)
+            .join(ArticleUrlKeywordMap)
+            .join(UrlKeyword)
+            .filter(UrlKeyword.keyword == URL_KEYWORD_TO_UPDATE)
+            .distinct()
+        ]
+    )
+
+    print(
+        f"Got articles with url_keyword '{URL_KEYWORD_TO_UPDATE}', total: {len(target_ids)}",
+    )
+
+    # Updating url_keyword new_topic mapping
+    # And the topics that were added based on that keyword.
+    if DELETE_ARTICLE_NEW_TOPICS:
+        new_topics_ids_to_delete = []
```

Comment:
is this precisely named? do we ever delete a `topic_id`?

Reply:
No, we don't delete topic_ids, but we delete any articles that contain the topics_ids that were associated with the keyword. 

I tried to name it 'topics_ids_to_delete_mappings', hopefully that makes more sense.